external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -330,11 +328,9 @@ public void ack(Object messageId) {
     public void fail(Object messageId) {
         final KafkaSpoutMessageId msgId = (KafkaSpoutMessageId) messageId;
         emitted.remove(msgId);
-        if (msgId.numFails() < maxRetries) {
-            msgId.incrementNumFails();
-            retryService.schedule(msgId);
-        } else { // limit to max number of retries
-            LOG.debug("Reached maximum number of retries. Message [{}] being marked as acked.", msgId);
+        msgId.incrementNumFails();
+        if (!retryService.schedule(msgId)) {
+            LOG.debug("Retry service indicated message should not be retried. Message [{}] being marked as acked.", msgId);
```

Comment:
I changed it because retryService can be any implementation here. It may decide not to schedule a message for other reasons than reaching the retry cap. I'll change it back if you still prefer the previous log message.